src/plugins/console/public/services/autocomplete.ts

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { createGetterSetter } from '@kbn/kibana-utils-plugin/public';
+import type { HttpSetup } from '@kbn/core/public';
+import type {
+  ClusterGetComponentTemplateResponse,
+  IndicesGetAliasResponse,
+  IndicesGetDataStreamResponse,
+  IndicesGetIndexTemplateResponse,
+  IndicesGetMappingResponse,
+  IndicesGetTemplateResponse,
+} from '@elastic/elasticsearch/lib/api/types';
+import { API_BASE_PATH } from '../../common/constants';
+import {
+  Alias,
+  DataStream,
+  Mapping,
+  LegacyTemplate,
+  IndexTemplate,
+  ComponentTemplate,
+} from '../lib/mappings';
+import { DevToolsSettings, Settings } from './settings';
+
+interface MappingsApiResponse {
+  mappings: IndicesGetMappingResponse;
+  aliases: IndicesGetAliasResponse;
+  dataStreams: IndicesGetDataStreamResponse;
+  templates: {
+    legacyTemplates: IndicesGetTemplateResponse;
+    indexTemplates: IndicesGetIndexTemplateResponse;
+    componentTemplates: ClusterGetComponentTemplateResponse;
+  };
+}
+
+export class AutocompleteInfo {
+  public readonly alias: Alias;
+  public readonly mapping: Mapping;
+  public readonly dataStream: DataStream;
+  public readonly legacyTemplate: LegacyTemplate;
+  public readonly indexTemplate: IndexTemplate;
+  public readonly componentTemplate: ComponentTemplate;
+  private http!: HttpSetup;
+  private pollTimeoutId: ReturnType<typeof setTimeout> | undefined;
+
+  constructor() {
+    this.alias = new Alias();
+    this.mapping = new Mapping();
+    this.dataStream = new DataStream();
+    this.legacyTemplate = new LegacyTemplate();
+    this.indexTemplate = new IndexTemplate();
+    this.componentTemplate = new ComponentTemplate();
+  }
+
+  public setup(http: HttpSetup) {
+    this.http = http;
+  }
+
+  public retrieve(settings: Settings, settingsToRetrieve: DevToolsSettings['autocomplete']) {
```

Comment:
can we somehow guarantee that more than one request will not be sent at the same time

Reply:
Plus I see risk that on 'get' you can faced with situation that data not loaded. This one also should be addressed somehow

Reply:
I agree. But I'd be satisfied with tracking this in an enhancement issue, since it looks like this behavior isn't new, right?

Reply:
Created a separate issue for this https://github.com/elastic/kibana/issues/132495